### ISSUE TITLE
Memory issue when using BentoML on Google Vertex IA (former Google IA plataform)

### ISSUE THREAD

**Issue by lucaslrolim** [bentoml/BentoML#1835]
I'm trying to follow the BentoML tutorial [1] to use Bento's API as a custom model deployment on Google Vertex IA (former Google IA Platform). Although the tutorial works, I'm facing some memory issues.

Independently of my machine type or volume of requests, each machine with my Bento container uses a maximum of 3 GB of memory , leading to many scalability problems. The image below shows memory usage of machine swith 8GB of memory.

![image](https://user-images.githubusercontent.com/9875473/131683748-f4b42537-ebdd-45aa-b2b1-0b6322a6e8e6.png)

I did some experiments changing the `mb_max_batch_size` parameters, and it seems to have some impact (before the maximum memory was 2.4 G).

My service definition is:

```
@artifacts([SklearnModelArtifact('model'), PickleArtifact('tfidf')])
class HotelDeduplicatorService(BentoService):
    @api(input=JsonInput(), mb_max_latency=3000, mb_max_batch_size=4000, batch=True)
    def predict(self, batch_input: List[JsonSerializable], tasks: List[InferenceTask]) -> dict:
...
```

[1] https://docs.bentoml.org/en/latest/deployment/google_cloud_ai_platform.html

**Comment by yubozhao** [comment]
@lucaslrolim Thanks for surfacing this issue! Is it still persisting?

**Comment by aarnphm** [comment]
A potential problem could be when the model server setup micro batching logics, without define `mb_max_batch_size`, the marshal layer will try to use most of the memory available. We will look into the issue and similar ones for the coming releases.

**Comment by aarnphm** [comment]
Hi @lucaslrolim, It has been a while. We have recently introduced a lot of changes for bentoml 1.0. Is it possible that you can try bentoml out again with `pip install -U --pre bentoml` and see if you still run into this issue.

**Comment by ssheng** [comment]
BentoML has released official 1.0.0. Could you please give it a try? Let us know if this problem persists.
